contracts/multisig-prover/src/encoding/sol/mod.rs

```diff
@@ -0,0 +1,136 @@
+use std::{array::TryFromSliceError, collections::BTreeMap};
+
+use axelar_encoding::types::{
+    CrossChainId, Message, Payload, PublicKey, Signature, Signer, WeightedSignature, WorkerSet,
+    U256,
+};
+
+use itertools::Itertools;
+use multisig::{msg::SignerWithSig, verifier_set::VerifierSet};
+
+use crate::error::ContractError;
+
+const ECDSA_COMPRESSED_PUBKEY_LEN: usize = 33; // this should be probably a public constant in axelar_encoding.
+const ED25519_PUBKEY_LEN: usize = 32; // this should be probably a public constant in axelar_encoding.
+
+type Result<T> = core::result::Result<T, ContractError>;
+
+pub fn to_worker_set(vs: &VerifierSet) -> Result<WorkerSet> {
+    let mut signers: BTreeMap<String, Signer> = BTreeMap::new();
+
+    vs.signers
+        .iter()
+        .try_for_each(|(address, signer)| -> Result<()> {
+            let enc_signer = signer.address.to_string();
+            let enc_pubkey = to_pub_key(&signer.pub_key)?;
+
+            let enc_weight = U256::from_be(to_u256_be(signer.weight.u128()));
+
+            signers.insert(
+                address.clone(),
+                Signer::new(enc_signer, enc_pubkey, enc_weight),
+            );
+            Ok(())
+        })?;
+
+    Ok(WorkerSet::new(
+        vs.created_at,
+        signers,
+        U256::from_be(to_u256_be(vs.threshold.u128())),
+    ))
+}
+
+fn to_pub_key(pk: &multisig::key::PublicKey) -> Result<PublicKey> {
+    Ok(match pk {
+        multisig::key::PublicKey::Ecdsa(hb) => {
+            PublicKey::new_ecdsa(hb.to_array::<ECDSA_COMPRESSED_PUBKEY_LEN>()?)
```

Comment:
minor: maybe use the full path to reference the "destination" type to make it easier to understand the undergoing type transformation? `axelar_encoding::types::PublicKey`

feel free to ignore if you disagree

Reply:
I am actually doing the opposite. I only use FQDN references for the multisig crate. But now that you point this out, i think your reasoning is better, as local readers will understand better the destination type as you said. I will apply this philosophy for all types references. Thank you !

Reply:
FIxed at 366d837323c0f65f18c2812ab1f19293f436108b